src/Agoda.Analyzers.Test/AgodaCustom/AG0018UnitTest.cs

```diff
@@ -0,0 +1,71 @@
+﻿// <copyright file="AG018Unittest.cs" company="Agoda Company Co., Ltd.">
+// AGODA ® is a registered trademark of AGIP LLC, used under license by Agoda Company Co., Ltd.. Agoda is part of Priceline (NASDAQ:PCLN)
+// </copyright>
+namespace Agoda.Analyzers.Test.AgodaCustom
+{
+    using Agoda.Analyzers.AgodaCustom;
+    using Agoda.Analyzers.Test.Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using NUnit.Framework;
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class AG0018UnitTest : DiagnosticVerifier
+    {
+        [Test]
+        public async Task AG0018_WhenCreateAMethodWhichReturnList_ShouldReturnAsAnInterface()
+        {
+            var code = $@"
+                        namespace Tests
+                        {{
+                            using System.Collections.Generic;
+                            using System.Collections.ObjectModel;
+                            public class IServerNamesProvider
+                            {{
+                                public List<double> LocalServer {{ get; set; }}
```

Comment:
Could this be split into multiple test cases? It's kind of big already and it's not obvious from the test which ones are allowed and which ones are disallowed.